migration/incremental/035-metric-table-refresh-view.sql

```diff
@@ -0,0 +1,9 @@
+ALTER TABLE _prom_catalog.metric ADD COLUMN view_refresh_interval INTERVAL DEFAULT NULL;
+
+-- Store rollup_id as column. This is helpful in knowing the retention duration for rollups (during Caggs retention process)
+-- registered in _prom_catalog.metric table.
+ALTER TABLE _prom_catalog.metric ADD COLUMN rollup_id BIGINT DEFAULT NULL;
+ALTER TABLE _prom_catalog.metric ADD CONSTRAINT fk_metric_rollup_id FOREIGN KEY (rollup_id) REFERENCES _prom_catalog.rollup(id);
```

Comment:
Thanks for pointing out. You are right, there shouldn't be any denormalization of rollup id here.

Reply:
Actually, just realized that this is more tricky than I thought, but in short, we need some `rollup` relation in metric table. We cannot use the id from metric table in some cases. Why?

Lets ask a question: How do we get metric id of Caggs from metric table that are not rollup? (for retention purpose) At the least, we need to store `is_rollup` in the table as a column.

Note: We cannot do `select metric_name from _prom_catalog.metric WHERE is_view AND id NOT IN (SELECT metric_id FROM _prom_catalog.metric_rollup)` since we register metric-rollups in the metric table, meaning we will get all those as answer. Why? Because metric table does not just contain metric, they contain caggs as well.

Reply:
I agree, `rollup_id` will be a better option here and simplify some parts of the code.